File: app/src/main/java/com/jeppsson/appexplore/UpdateReceiver.java
package com.jeppsson.appexplore;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class UpdateReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        String action = intent.getAction();

        if (Intent.ACTION_PACKAGE_FULLY_REMOVED.equals(action)
                || Intent.ACTION_PACKAGE_ADDED.equals(action)) {
            PackageScannerService.enqueueWork(context, intent);
        }
    }
}

File: app/src/main/java/com/jeppsson/appexplore/UpdateReceiver.kt
package com.jeppsson.appexplore

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class UpdateReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        val action = intent.action

        if (Intent.ACTION_PACKAGE_FULLY_REMOVED == action
                || Intent.ACTION_PACKAGE_ADDED == action) {
            PackageScannerService.enqueueWork(context, intent)
        }
    }
}
